Samples/ThreatIntelligenceAPI/IndicatorConverter.py

```
@@ -0,0 +1,58 @@
+from collections import defaultdict
+import datetime
+from datetime import timedelta
+import json
+
+class IndicatorConverter: 
+    """A class that handles converting events to Stix Indicators for use in the Threat Intelligence Indicator API
+    """
+
+    @staticmethod
+    def _generate_indicator(event) -> defaultdict:
+        """
+        Sets up required indicator values, including: "type", "spec_version", "id", "created", "modified", "pattern", "pattern_type", and "valid_from"
+        Sets up optional indicator values, which are not necessary to send an indicator but can be populated if the information is given
+        Input:  indicator - the list to add required information to. 
+                event - the event itself that is being parsed
+        Output: defaultdict of all required indicator information. 
+        """
+        indicator = defaultdict(list) 
+        for attr in event['Attribute']: # parses pattern
+            if attr['type'] == 'ip-src': 
+                indicator['pattern'] = 'ipv4-addr:value = \'' + attr['value'] + '\''
+            elif attr['type'] == 'domain': 
+                indicator['pattern'] = 'domain-name:value = \'' + attr['value'] + '\''
+            else: 
+                indicator['pattern'] = attr['type'] + ':value = \'' + attr['value'] + '\''
+        indicator['spec_version'] = '2.1' # parses spec_version
+        indicator['id'] = event.get("uuid", "") # parses id
```

We should throw exception if one of required fields is missing